--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,7 +56,7 @@
     VERSION_HEADER ${CMAKE_CURRENT_BINARY_DIR}/neochat-version.h
 )
 
-find_package(Qt6 ${QT_MIN_VERSION} NO_MODULE COMPONENTS Core Quick Gui QuickControls2 Multimedia Svg TextToSpeech WebView)
+find_package(Qt6 ${QT_MIN_VERSION} NO_MODULE COMPONENTS Core Quick Gui QuickControls2 Multimedia Svg WebView)
 set_package_properties(Qt6 PROPERTIES
     TYPE REQUIRED
     PURPOSE "Basic application components"
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -33,9 +33,14 @@
     identityserverhelper.h
     models/commonroomsmodel.cpp
     models/commonroomsmodel.h
+)
+
+if(NOT SAILFISHOS)
+add_library(neochat STATIC
     texttospeechhelper.h
     texttospeechhelper.cpp
 )
+endif()
 
 set_source_files_properties(qml/OsmLocationPlugin.qml PROPERTIES
     QT_QML_SINGLETON_TYPE TRUE
@@ -189,7 +194,6 @@
     Qt::Multimedia
     Qt::Network
     Qt::QuickControls2
-    Qt::TextToSpeech
     KF6::I18n
     KF6::Kirigami
     KF6::Notifications
