diff --git a/CMakeLists.txt b/CMakeLists.txt
index 562c254a2..0e184797d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,7 +89,15 @@ if(ANDROID)
     )
 else()
     find_package(Qt6 ${QT_MIN_VERSION} COMPONENTS Widgets)
-    find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS QQC2DesktopStyle KIO WindowSystem StatusNotifierItem Crash)
+    if (SAILFISHOS)
+        find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS KIO WindowSystem)
+        find_package(QQC2BreezeStyle ${KF_MIN_VERSION} REQUIRED)
+        set_package_properties(QQC2BreezeStyle PROPERTIES
+            TYPE RUNTIME
+        )
+    else()
+        find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS QQC2DesktopStyle KIO WindowSystem StatusNotifierItem Crash)
+    endif()
     find_package(KF6SyntaxHighlighting ${KF_MIN_VERSION} REQUIRED)
     set_package_properties(KF6QQC2DesktopStyle PROPERTIES
         TYPE RUNTIME
