diff --git a/CMakeLists.txt b/CMakeLists.txt
index 562c254a2..2bad76954 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,12 +89,16 @@ if(ANDROID)
     )
 else()
     find_package(Qt6 ${QT_MIN_VERSION} COMPONENTS Widgets)
-    find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS QQC2DesktopStyle KIO WindowSystem StatusNotifierItem Crash)
-    find_package(KF6SyntaxHighlighting ${KF_MIN_VERSION} REQUIRED)
-    set_package_properties(KF6QQC2DesktopStyle PROPERTIES
-        TYPE RUNTIME
-    )
-    ecm_find_qmlmodule(org.kde.syntaxhighlighting 1.0)
+    if(NOT SAILFISHOS)
+        find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS QQC2DesktopStyle KIO WindowSystem StatusNotifierItem Crash)
+        find_package(KF6SyntaxHighlighting ${KF_MIN_VERSION} REQUIRED)
+        set_package_properties(KF6QQC2DesktopStyle PROPERTIES
+            TYPE RUNTIME
+        )
+        ecm_find_qmlmodule(org.kde.syntaxhighlighting 1.0)
+    else()
+        find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS KIO WindowSystem)
+    endif()
 
     find_package(ICU 61.0 COMPONENTS uc)
     set_package_properties(ICU PROPERTIES
diff --git a/src/messagecontent/CMakeLists.txt b/src/messagecontent/CMakeLists.txt
index bd93c6449..d6ac2b2fc 100644
--- a/src/messagecontent/CMakeLists.txt
+++ b/src/messagecontent/CMakeLists.txt
@@ -102,6 +102,6 @@ target_link_libraries(MessageContent PRIVATE
     LibNeoChat
 )
 
-if(NOT ANDROID)
+if(NOT ANDROID AND NOT SAILFISHOS)
     target_link_libraries(MessageContent PUBLIC KF6::SyntaxHighlighting)
 endif()
diff --git a/src/messagecontent/models/messagecontentmodel.cpp b/src/messagecontent/models/messagecontentmodel.cpp
index 194c661c5..21fc91799 100644
--- a/src/messagecontent/models/messagecontentmodel.cpp
+++ b/src/messagecontent/models/messagecontentmodel.cpp
@@ -21,7 +21,7 @@
 #include <KLocalizedString>
 #include <Kirigami/Platform/PlatformTheme>
 
-#ifndef Q_OS_ANDROID
+#if !defined(Q_OS_ANDROID) && !defined(SAILFISHOS)
 #include <KSyntaxHighlighting/Definition>
 #include <KSyntaxHighlighting/Repository>
 #endif
@@ -590,7 +590,7 @@ QList<MessageComponent> MessageContentModel::componentsForType(MessageComponentT
             if (!m_isReply) {
                 auto fileTransferInfo = m_room->cachedFileTransferInfo(event.first);
 
-#ifndef Q_OS_ANDROID
+#if !defined(Q_OS_ANDROID) && !defined(SAILFISHOS)
                 Q_ASSERT(roomMessageEvent->content() != nullptr && roomMessageEvent->has<EventContent::FileContent>());
                 const QMimeType mimeType = roomMessageEvent->get<EventContent::FileContent>()->mimeType;
                 if (mimeType.name() == u"text/plain"_s || mimeType.parentMimeTypes().contains(u"text/plain"_s)) {
